### PR TITLE
fix: Proper type-hinting & docstring of `Member.move_to`  and `Member.edit`

### PR DESCRIPTION
**Pull request by yoggys**
## Summary

Fixed type-hinting and docstring for `Member.move_to` as channel is excepted to be `channel: Optional[:class:VoiceChannel]` and `Member.edit`.

## Information

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,11 @@ possible (see our [Version Guarantees] for more info).
 
 These changes are available on the `master` branch, but have not yet been released.
 
+### Fixed
+
+- Fixed the type-hinting of `Member.move_to` to reflect actual behavior.
+-
```
Comment (Lulalaby) [inline CHANGELOG.md:16]:
```suggestion
```

**Review by Lulalaby — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (yoggys) [reply CHANGELOG.md]: Ahaha I missed it

**Review by Lulalaby — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,11 @@ possible (see our [Version Guarantees] for more info).
 
 These changes are available on the `master` branch, but have not yet been released.
 
+### Fixed
+
+- Fixed the type-hinting of `Member.move_to` to reflect actual behavior.
+- ([#2386](https://github.com/Pycord-Development/pycord/pull/2386))
```
Comment (plun1331) [inline CHANGELOG.md:16]:
```suggestion
  ([#2386](https://github.com/Pycord-Development/pycord/pull/2386))
```

**Review by plun1331 — CHANGES_REQUESTED** [review]

**Comment by yoggys** [comment]
@plun1331 I've also updated `Member.edit` docstring

**Review by Lulalaby — DISMISSED** [review]

**Comment by yoggys** [comment]
should be all related to `StageChannel`, i've also noticed one typing improvement in `afk_channel` as it is always `VoiceChannel`

**Review by Lulalaby — APPROVED** [review]

**Review by JustaSqu1d — APPROVED** [review]

**Pull request merged**